Reformat drawio, rework diagrams and overview page. (#65)
wireapp/wire-docs: src/understand/overview.md

In a simplified way, the server components for Wire involve the following:

![image](img/architecture-server-simplified.svg)

> The Wire clients (such as the Wire app on your phone) connect either directly (or via a load balancer) to the “Wire Server”. By “Wire Server” we mean multiple API server components that connect to each other, and which also connect to a few databases. Both the API components and the databases are each in a “cluster”, ensuring copies of the same program are available multiple times. This allows any one component to fail without users noticing that there is a problem (also called “high-availability”).

Reviewer: ```suggestion
The Wire clients (such as the Wire app on your phone) connect either directly or through a load balancer to the **Wire Server**.  The term **Wire Server** refers to a set of API server components that work together and also communicate with various databases. Both the API components and the databases are each deployed in a **cluster**, a setup where multiple instances of the same service run in parallel. This design ensures that if one instance fails, others continue to serve requests without disruption. This fault-tolerant setup is known as high availability, and it helps ensure a seamless experience for users even during component failures.
```

## Architecture and networking

### Client communications
Users use Wire Clients, either installed with the wire application locally, or by using the web version of the wire application. Let's take a look at the wire backend, from the perspective of a client device:
![image](img/architecture-client_communications.svg)

The above graph gives us a rough breakdown of which domains are used for which parts of the platform.

#### Wire Web Applications

> The Wire web application is started by a web browser opening up https://webapp.example.com/ . Similarly, Teams in Wire are managed by loading the Team Management application, reachable at https://teams.example.com/ . Both of these web applications are static, which is to say, the application that is loaded uses the Wire API to perform all actions, and has no "state" in the backend. the 'account' application is not meant for users to directly access, and contains pages to manage password resets, and other special pages not a part of team settings or the Wire Webapp.

Reviewer: ```suggestion
The Wire web application is started by a web browser opening up https://webapp.example.com/ . Similarly, Teams in Wire are managed by loading the Team Management application, reachable at https://teams.example.com/ . Both of these web applications are static, which is to say, the application that is loaded uses the Wire API to perform all actions, and has no "state" in the backend. The 'account' application is not meant for users to directly access, and contains pages to manage password resets, and other special pages outside of the team settings or the Wire Webapp.
```

#### Wire API Access
All actions in the wire applications require wire API access, which is generally granted when a user logs into a wire client device (including webapp). There are no "long lived" connections in the API, when a client is using the application, they perform their API call, then the connection ends. 

#### Wire Websocket Notifications
As wire is an interactive application, from time to time, we need to tell the client something has happened. This involves holding open a "long lived" connection to the backend, via the WebSockets API over HTTP. 

#### Asset Retrieval

> Files, Voice Messages, Pictures, and other items users upload to the wire backend for distribution are referred to as 'assets' in wire. Asset requests are handled on the "https://assets.example.com/" domain, which is separate from the rest of wire, so that customers have the option of using cacheing (cloudflare, etc) services to speed up their downloads.

Reviewer: ```suggestion
Files, Voice Messages, Pictures, and other items users upload to the wire backend for distribution are referred to as 'assets' in wire. Asset requests are handled on the "https://assets.example.com/" domain, which is separate from the rest of wire. This separation allows customers to use caching services (such as Cloudflare) to accelerate downloads and improve performance.
```

#### Calling
In general, both of wire's calling services have two phases of communicating with them: there's the signaling phase, which creates, reserves, and communicates about the call, then there is the active phase, where audio/video data is being transfered in-between the participants. Part of the signaling phase of all calls is performed by messages over the Wire Messaging Platform, in order to inform participants about the existence and location of a call.

##### Conference Calling

> Conference calling in Wire is managed a dedicated service, named "SFT". This service uses HTTPS to perform the signaling parts of setting up a call, but uses a proprietary deritive of the TURN protocol over UDP to actually transfer calling content. This is why it goes both to a load balancer (for HTTPS), and to a firewall (to forward on UDP packets).

Reviewer: ```suggestion
Conference calling in Wire is a managed dedicated service, named "SFT". This service uses HTTPS to perform the signaling parts of setting up a call, but uses a proprietary derivative of the TURN protocol over UDP to actually transfer calling content. This is why it goes both to a load balancer (for HTTPS), and to a firewall (to forward on UDP packets).
```

##### Personal Calling
Individual calling between pairs of participants in the wire application is managed by a version of https://github.com/coturn/coturn , which wire has extended for denial of service hardening. This service does not use HTTPS at any point, but does speak the TURN protocol, only deviating from the published standard in the area of authentication.

#### Mobile Notifications

> At this point, you might be wondering "what's that Amazon thing up in the corner?". For mobile clients (AKA, cellphones), Wire utilizes Google and Apple push notification services to send a message to the client devices when they need to go check the backend for messages. This is because keeping a "long lived" connection to the backend causes a major battery draw on android, and is simply not possible on apple devices. If complete secrecy is required, the android clien is permitted to go into a "websocket-only" mode, where it will use the websocket instead of the cell phone network. This, of course, drains the battery, as the messages required to keep the "long lived" connection going require regular contact to your backend, and require "waking up" the mobile device regularly, preventing it from going into deeper power savings modes.

Reviewer: ```suggestion
At this point, you might be wondering "what's that Amazon thing up in the corner?". For mobile clients (AKA, cellphones), Wire utilizes Google and Apple push notification services to send a message to the client devices when they need to go check the backend for messages. This is because keeping a "long lived" connection to the backend causes a major battery draw on android, and is simply not possible on apple devices. If complete secrecy is required, the android client is permitted to go into a "websocket-only" mode, where it will use the websocket instead of the cell phone network. This, of course, drains the battery, as the messages required to keep the "long lived" connection going require regular contact to your backend, and require "waking up" the mobile device regularly, preventing it from going into deeper power savings modes.
```
